src/Dialogs/ChooseAppDialog.vala

```diff
@@ -48,30 +50,30 @@ class PF.ChooseAppDialog : Object {
 
         check_default.show ();
 
-        dialog.get_content_area ().add (check_default);
-
-        dialog.show ();
+        dialog.get_content_area ().append (check_default);
     }
 
     public AppInfo? get_app_info () {
```

Comment:
This will just return the null AppInfo since we're no longer blocking with `run()`. So this needs to make sure we're actually blocking the function from returning until we get that response